portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -55,21 +70,43 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = getParameter(request, "text-input", "");
+    String comment = getStringParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
     // Add comment entity to Datastore
     Entity commentEntity = new Entity(COMMENT_ENTITY);
     commentEntity.setProperty(COMMENT_CONTENT_PROPERTY, comment);
+    commentEntity.setProperty(COMMENT_TIMESTAMP_PROPERTY, timestamp);
     datastore.put(commentEntity);
     // Redirect to about me page on post
     response.sendRedirect("/index.html");
   }
 
-  /** Returns the request parameter, or the default value if not specified */
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+  /** Returns the request parameter (for Strings), or the default value if not specified */
+  private String getStringParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
     return value == null ? defaultValue : value;
   }
 
+  /** Get number of comments to display from request */ 
+  private int getNumCommentsParameter(HttpServletRequest request) {
+    String numCommentsString = request.getParameter(NUM_COMMENTS_PARAMETER);
+
+    // Convert string value to int
+    int numComments;
+    try {
+      numComments = Integer.parseInt(numCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numCommentsString);
+      return -1;
```

Comment:
If I was going to future-proof this, I would return `Integer.MIN_VALUE`. This is well below and reasonable input usage and it allows a callsite to decide what values it's willing to accept. This is just a helper method to extract the parameter.